app/models/bot/smooch.rb

```diff
@@ -766,6 +754,26 @@ def self.save_message(message_json, app_id, author = nil)
     self.send_report_to_user(message['authorId'], message, pm, message['language'])
   end
 
+  def self.create_project_media_from_message(message)
+    pm = case message['type']
+         when 'text'
+           self.save_text_message(message)
+         when 'image'
+           self.save_media_message(message)
+         when 'video'
+           self.save_media_message(message, 'video')
+         when 'audio'
+           self.save_media_message(message, 'audio')
+         when 'file'
+           message['mediaType'] = self.detect_media_type(message)
+           m = message['mediaType'].to_s.match(/^(image|video|audio)\//)
+           m.nil? ? return : self.save_media_message(message, m[1])
+         else
+           return
+         end
```

Comment:
I suggest we create a `save_message(message)` and get rid of this whole `case..when` block... because this way we can just call `save_message(message)`. Also, in most cases the `case` value is the same as the `type` parameter for `save_media_message`.